internal/util/prometheus.go

```diff
@@ -0,0 +1,74 @@
+package util
```

Comment:
With so many types and functions deifned specifically for the metrics bounded context, would it make sense to extract it to a separate package?

Reply:
@mflendrich which types and functions you are referring to ? The prometheus module is really small - several functions and types.

Reply:
my 2cents would be to leave this here for now and focus on functionality: we have the benefit that everything is `internal/` now, anything contributed in `internal/` is much less contentious to move around later than if it were public.

Reply:
The `util` package is for general purpose "utilities".

The stuff that we're adding in this PR:
- a type that stores prometheus metrics for the business domain of the controller
- a constructor for the above
- type and const definitions that only make sense in the prometheus context

has the following two features:
- very clearly belongs to a very specific bounded context "metrics"
- is not reusable (because the metrics defined in this code are tightly coupled with the system they'll be used by)

and therefore is not an `util`.

>  we have the benefit that everything is internal/ now, anything contributed in internal/ is much less contentious to move around later than if it were public.

I agree, but this shouldn't mean that we do not modularize the code appropriately.

Therefore my suggestion is that `util` is not a good package for these bits mentioned above - I suggest moving them to a separate package, for example `/internal/metrics`.

Reply:
👍  agree, internal/metrics does bring better clarity, do it.

Reply:
changed 9464c949d7d461c4c1cc57225499c2c8ff471599